aten/src/THC/THCTensorMathPointwise.cuh

```diff
@@ -779,5 +779,49 @@ struct TensorBitXorOp {
   }
 };
 
+template <typename real, typename accreal>
+struct TensorDigammaOp {
+  __device__ __forceinline__ void
+  operator()(real* out, real* in) {
+    static accreal PI = 3.14159265358979323846;
+    accreal x = ScalarConvert<real, accreal>::to(*in);
+    accreal result = 0;
```

Comment:
You probably don't want to do all this math in `accreal`. This will have a huge perf penalty on non-Tesla cards

Reply:
Is there a way to use `real` for everything but `halffloat` and use `float` for `halffloat`? I originally implemented the all-`real` solution you suggest but there were a bunch of `halffloat` errors I didn't understand and template specialization seemed overly complex.

Reply:
One way would be to do some template magic a'la
```cpp
using compute_type = typename std::conditional<std::is_same<real, half>::value, accreal, real>::type;
```

Reply:
Done.